app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,24 @@ def show
     @user = User.find(params[:user_id])
     @post = @user.posts.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = @current_user.posts.build(post_params)
+
+    if @post.save
+      redirect_to user_post_path(user_id: @current_user.id, id: @post.id)
+    else
+      render :new
+    end
+  end
```

Comment:
- [ ] I couldn't create a post using the form which is being rendered using the new action. The action is rolled back without any error message. Kindly fix this issue and please for a better user experience, add buttons for adding a post, comment, and like.

![rollback](https://github.com/ab-noori/Blog-App/assets/53124548/4af5078b-1ac1-4030-950c-d5e340172881)

Reply:
**Hi @elisha2kyakpo1 👋,**
Thank you so much 🙏 for reviewing my pull request.
Your nice feedback is really appreciated 🙏.
I have checked and made them fix 🎯.

![thank-you-gif-22](https://github.com/ab-noori/bookstore-cms/assets/111179533/0848000e-52ce-440c-a114-6c7f859a0eaa)